client/features/authSlice.tsx

```diff
@@ -0,0 +1,30 @@
+import { createSlice } from '@reduxjs/toolkit'
+import type { PayloadAction } from '@reduxjs/toolkit'
+import type { User } from '../../src/types/custom'
+import type { RootState } from '../app/store'
+
+type AuthState = {
+  user: User | null
+  token: string | null
+}
+
+const authSlice = createSlice({
+  name: 'auth',
+  initialState: { user: null, token: null } as AuthState,
```

Comment:
Do not recommending cast types when it's avoidable - prefer to use an interface, and create initial state separately (there are examples of this in the Redux Quick Start docs for TypeScript)

Usually casting is not recommended unless absolutely necessary. Some useful discussion here, the top-rated answers are pretty helpful: https://www.reddit.com/r/typescript/comments/wd3f7j/should_i_avoid_casting_types/

```suggestion
interface AuthState = {
  user: User | null
  token: string | null
}

const initialState: AuthState = { 
  user: null,
  token: null
};

const authSlice = createSlice({
  name: 'auth',
  initialState,
```

Reply:
More info on the difference between types and interfaces: https://www.typescriptlang.org/play#example/types-vs-interfaces

Reply:
Thanks for catching that. This was a boilerplate I think. Yeah, I try to avoid typecasting.